src/Reexport.jl

```diff
@@ -1,28 +1,42 @@
 module Reexport
 
-macro reexport(ex)
-    isa(ex, Expr) && (ex.head == :module ||
-                      ex.head == :using ||
-                      (ex.head == :toplevel &&
-                       all(e->isa(e, Expr) && e.head == :using, ex.args))) ||
+macro reexport(ex::Expr)
+    esc(reexport(__module__, ex))
+end
+
+reexport(m::Module, l::LineNumberNode) = l
+
+function reexport(m::Module, ex::Expr)
+    # unpack any macros
+    ex = macroexpand(m, ex)
+    # recursively unpack any blocks
+    if ex.head == :block
+        return Expr(:block, map(e -> reexport(m, e), ex.args)...)
+    end
+
+    Meta.isexpr(ex, [:module, :using, :import]) ||
+        Meta.isexpr(ex, :toplevel) && all(e -> isa(e, Expr) && e.head == :using, ex.args) ||
         error("@reexport: syntax error")
 
     if ex.head == :module
         modules = Any[ex.args[2]]
         ex = Expr(:toplevel, ex, :(using .$(ex.args[2])))
-    elseif ex.head == :using && all(e->isa(e, Symbol), ex.args)
+    elseif ex.head == :using && all(e -> isa(e, Symbol), ex.args)
```

Comment:
@ararslan I think this branch can never be reached? In fact, I removed it and all tests still pass. Do you know more?

Reply:
Ah, you're right, good catch. This package is very old—it was written for Julia 0.2, I believe—and the way some of these expressions are structured internally has since changed but not all of the code here has been updated to match. (I thought it had been but clearly not!) AFAICT, this branch would have accepted `:(using A)` and `:(using A.B)` in earlier Julia versions. As you've noticed, it seems now to be dead code.